### PR TITLE
Miscellaneous small updates for 2.0

### PR DESCRIPTION
**Pull request by ctb**
Currently tackling:
- #1097 - make --csv and -t default
- #1130 - update citation information for semi-streaming paper
- #1009 - update Python version info
- #820 - remove parse_description from everything
- #274 - remove perf_metrics and trace_logger references

### CODE REVIEW

**Comment by mr-c** [comment]
This is great!

**Comment by ctb** [comment]
- [ ] Is it mergeable?
- [ ] Did it pass the tests?
- [ ] If it introduces new functionality in scripts/ is it tested?
  Check for code coverage with `make clean diff-cover`
- [ ] Is it well formatted? Look at `make pep8`, `make diff_pylint_report`,
  `make cppcheck`, and `make doc` output. Use `make format` and manual
  fixing as needed.
- [ ] Did it change the command-line interface? Only additions are allowed
  without a major version increment. Changing file formats also requires a
  major version number increment.
- [ ] Is it documented in the ChangeLog?
  http://en.wikipedia.org/wiki/Changelog#Format
- [ ] Was a spellchecker run on the source code and documentation after
  changes were made?
- [ ] Is the Copyright year up to date?

**Comment by ctb** [comment]
This is ready for preliminary review, @mr-c. I have some CITATION stuff to add but otherwise done with planned work.

**Comment by ctb** [comment]
(and it passes on py3)

File: khmer/khmer_args.py [hunk 1]
```diff
@@ -226,7 +226,7 @@ def add_threading_args(parser):
     'software': 'MR Crusoe et al., '
     '2014. http://dx.doi.org/10.6084/m9.figshare.979190',
```
Comment (mr-c) [inline khmer/khmer_args.py]: Shall we publish a draft to get a DOI with F1000?

**Comment by ctb** [comment]
Oof, diff-cover reports:
```
khmer/thread_utils.py (100%)
scripts/abundance-dist-single.py (0.0%): Missing lines 85,87,119,170
scripts/abundance-dist.py (0.0%): Missing lines 110,112,123
scripts/count-median.py (0.0%): Missing lines 89,91,93,101
scripts/count-overlap.py (0.0%): Missing lines 65,67,84
scripts/extract-long-sequences.py (0.0%): Missing lines 48
scripts/extract-paired-reads.py (0.0%): Missing lines 127
scripts/extract-partitions.py (0.0%): Missing lines 37
scripts/fastq-to-fasta.py (0.0%): Missing lines 46
scripts/filter-abund-single.py (0.0%): Missing lines 96
scripts/interleave-reads.py (0.0%): Missing lines 107-108
scripts/load-into-counting.py (0.0%): Missing lines 136-138
scripts/normalize-by-median.py (0.0%): Missing lines 361
scripts/readstats.py (0.0%): Missing lines 133
scripts/sample-reads-randomly.py (0.0%): Missing lines 134
scripts/split-paired-reads.py (0.0%): Missing lines 133
scripts/trim-low-abund.py (0.0%): Missing lines 168,264
```

**Comment by ctb** [comment]
All the missing code coverage turns out to be a bug in diff-cover; we're all good on the coverage side.

File: khmer/khmer_args.py (hunk 1 above)
Comment (ctb) [inline khmer/khmer_args.py]: Sure, but no particular hurry.

**Comment by ctb** [comment]
Ready for review/merge, @mr-c.

**Comment by ctb** [comment]
The tests are failing on pylint stuff that I didn't change.

**Comment by mr-c** [comment]
LGTM, Thanks!

**Pull request merged**